src/main/java/tutorlink/grade/Grade.java

```diff
@@ -23,6 +23,9 @@ public Component getComponent() {
     }
 
     public double getScore() {
+        if(component!= null && component.getMaxScore() < score){
+            score = component.getMaxScore();
+        }
```

Comment:
Ok this is a reasonable way to deal with the new score being lower than the max score, but we must include this in the UG.